File: challenge-load-test-chat/test-by-locust/src/tasks/update_message.py
from locust import task, TaskSet
from endpoints import Endpoints

class UpdateMessageTask(TaskSet):
    @task
    def update_message(self):
        chat_id = getattr(self.user, 'chat_id', None)
        message_id = getattr(self.user, 'message_id', None)
        if chat_id and message_id:
            self.client.put(Endpoints.MESSAGE_DETAIL_UPDATE.format(chat_id=chat_id, message_id=message_id), json={"sender": "locust", "content": "updated"})

File: challenge-load-test-chat/test-by-k6/src/tasks/update-message.ts
import http from 'k6/http';
import { check } from 'k6';
import { Endpoints } from '../endpoints';

export function updateMessage(baseUrl: string, chatId: string, messageId: string) {
  const payload = JSON.stringify({ sender: 'k6-user', content: 'Updated by k6' });
  const params = { headers: { 'Content-Type': 'application/json' } };
  const res = http.put(`${baseUrl}${Endpoints.MESSAGE_DETAIL_UPDATE(chatId, messageId)}`, payload, params);
  check(res, { 'update message status is 200': (r) => r.status === 200 });
  return res;
}
